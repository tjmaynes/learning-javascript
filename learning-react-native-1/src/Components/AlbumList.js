import React, { Component } from 'react';
import { ScrollView, Text } from 'react-native';
import AlbumDetail from './AlbumDetail';

import axios from 'axios';

class AlbumList extends Component {
  state = { albums: [] };
  
  componentWillMount() {
    axios.get('https://rallycoding.herokuapp.com/api/music_albums')
      .then(results => this.setState({ albums: results.data }));
  }

  renderAlbums() {
    return this.state.albums.map(album => <AlbumDetail key={album.title} album={album}/>);
  }

  render() {
    return (
      <ScrollView>
	{this.renderAlbums()}
      </ScrollView>
    );
  };
};

export default AlbumList;
