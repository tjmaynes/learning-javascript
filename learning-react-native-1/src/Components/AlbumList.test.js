import React from 'react';
import AlbumList from './AlbumList';

import renderer from 'react-test-renderer';

describe('AlbumListComponent', () => {
  it('renders as expected', () => {
    const rendered = renderer.create(<AlbumList />).toJSON();
    expect(rendered).toBeTruthy();
    expect(rendered).toMatchSnapshot();
  });
});
