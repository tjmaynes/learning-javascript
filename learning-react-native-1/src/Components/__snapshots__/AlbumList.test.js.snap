// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumListComponent renders as expected 1`] = `<View />`;
